src/executables/master_heartbeat.c

```diff
@@ -4394,10 +4394,14 @@ hb_resource_receive_changemode (CSS_CONN_ENTRY * conn)
     {
     case HA_SERVER_STATE_ACTIVE:
       proc->state = HB_PSTATE_REGISTERED_AND_ACTIVE;
+      /* When the ATS is set to active, then the server state is set to ready.
+       * Temporary setting to make understand easier */
+      proc->server_state = HB_SERVER_STATE_READY;
       break;
 
     case HA_SERVER_STATE_TO_BE_ACTIVE:
       proc->state = HB_PSTATE_REGISTERED_AND_TO_BE_ACTIVE;
+      proc->server_state = HB_SERVER_STATE_RECOVERY;
```

Comment:
This is an example for clarification purposes and will be removed after the PR review.

After the implementation of state transitions is complete, assigning `server_state` for each state transition will be added.

Reply:
I can't see the state transition issue. Is this in the next epic or on me? In my epic, PS doesn't know its state actually.

Reply:
state transition will be implemented in the next epic. Currently, I tried to demonstrate an example where the `server_state` changes when the state changes. Of course, these modifications will be discarded in this PR.

Reply:
hmm.. why is the state-setting issue bundled in the failover epic? It seems to be about the not initialization. Or the "next epic" is another epic?

Reply:
Hmm...I apologize for the confusion caused by using the term "next epic." 

I'm planning to address this issue in the epic that handles the recovery of TS and PS (http://jira.cubrid.org/browse/LETS-708). Since it involves further refining the service preparation phase (recovery), it seems that LETS-708 will be the appropriate place to actually setting these states.

Reply:
hmmmm... Doesn't it mean that thie PR/issue belongs to the "next" epic? Do we really need to define these in advance in this epic? I think issues in an epic should be concluded by themselves. Let me have your opinion.

Reply:
I will close this PR and open again in the "next" epic (http://jira.cubrid.org/browse/LETS-708).